File: Game.py
import random

rand = random.randint(1, 20)
attempts = 1
money = 500

print("Welcome to the Guessing Game! You have £500!")
confirmation = input("\nDo you want to play the Guessing Game? ")
if confirmation == "Yes":
  game = True
  while money >= 500 and game:
    if attempts <= 5:
      guess = int(input("Enter a number between 1 and 20: "))
      if guess == rand:
        money -= 500
        attempts += 1
        print("Congratulations! You guessed it in " + str(attempts - 1) + " attempts!")
        money += 1000
        print("You now have £" + str(money) + "\n")
        continueGame = input("Do you want to continue playing the game? ")
        if continueGame == "Yes":
          rand = random.randint(1, 20)
          attempts = 1
        else:
          print("See you later!")
          game = False
      elif guess > rand:
        attempts += 1
        print("Your guess was bigger than the number!")
      elif guess < rand:
        attempts += 1
        print("Your guess was smaller than the number!")
    else:
      money -= 500
      print("Sorry you used up all your attempts, the right number was " + str(rand))
      continueGame2 = input("Do you want to continue playing the game? ")
      if continueGame == "Yes":
        rand = random.randint(1, 20)
        attempts = 1
      else:
        print("See you later!")
        game = False

File: Game.cpp
#include <iostream>
#include <ctime>
#include <cstdlib>
using namespace std;

int main()
{
    srand(time(0));

    int guess;
    int attempts = 1;
    int money = 500;
    int randNum = rand() % 20 + 1;
    string confirmation;
    string continueGame;
    string continueGame2;
    bool game;

    cout << "Welcome to the Guessing Game!";
    cout << "\nDo you want to play the Guessing Game? ";
    cin >> confirmation;

    if (confirmation == "Yes")
    {
        game = true;
        while ((money >= 500) && (game))
        {
            if (attempts <= 5)
            {
                cout << "Enter a number between 1 and 20: ";
                cin >> guess;
                if (guess == randNum)
                {
                    money = money - 500;
                    attempts++;
                    cout << "Congratulations! You guessed it in " << attempts - 1 << " attempts!";
                    money = money + 1000;
                    cout << "You now have £" << money << "\n";
                    cout << "Do you want to continue playing the game? ";
                    cin >> continueGame;
                    if (continueGame == "Yes")
                    {
                        randNum = rand() % 20 + 1;
                        attempts = 1;
                    }
                    else
                    {
                        cout << "See you later!";
                        game = false;
                    }
                }
                else if (guess > randNum)
                {
                    attempts++;
                    cout << "Your guess was bigger than the number!\n";
                }
                else if (guess < randNum)
                {
                    attempts++;
                    cout << "Your guess was smaller than the number!\n";
                }
            }
            else
            {
                money = money - 500;
                cout << "Sorry you used up all your attempts, the right number was " << randNum << "\n";
                cout << "Do you want to continue playing the game? ";
                cin >> continueGame2;
                if (continueGame2 == "Yes")
                {
                    randNum = rand() % 20 + 1;
                    attempts = 1;
                }
                else
                {
                    cout << "See you later!";
                    game = false;
                }
            }
        }
    }
}
